src/js/libcs/OpImageDrawing.js

```diff
@@ -388,3 +388,167 @@ evaluator.cameravideo$0 = function() {
+            if (erg.ctype === 'boolean') {
+                interpolate = (erg.value);
+            }
+        }
+
+        if (modifs.repeat !== undefined) {
+            erg = evaluate(modifs.repeat);
+            if (erg.ctype === 'boolean') {
+                repeat = (erg.value);
+            }
+        }
+    }
+    handleModifs();
+
+    if (!img) return nada;
+
+    var w = img.width;
+    var h = img.height;
+
+    var w0 = evaluateAndHomog(args[0]);
+    var w1 = evaluateAndHomog(args[1]);
+    var v0 = evaluateAndHomog(List.realVector([0, h, 1]));
+    var v1 = evaluateAndHomog(List.realVector([w, h, 1]));
+
+    if (w0 === nada || w1 === nada || p === nada) return nada;
+
+    //create an orientation-reversing similarity transformation that maps w0->v0, w1->v1
+    var ii = List.ii;
+    var jj = List.jj;
+
+    var m1 = eval_helper.basismap(v0, v1, ii, jj); //interchange I and J,
+    var m2 = eval_helper.basismap(w0, w1, jj, ii); //see Thm. 18.4 of Perspectives on Projective Geometry
+    var p = evaluateAndHomog(args[3]);
+    var coord = eval_helper.extractPoint(General.mult(m1, General.mult(List.adjoint3(m2), p)));
+
+    if (!coord.ok) return nada;
+
+    if (interpolate) {
+        coord.x -= 0.5; //center of pixels are in the middle of them.
+        coord.y -= 0.5; //Now pixel-centers have wlog integral coordinates
+    }
+
+    if (repeat) {
+        coord.x = (coord.x % w + w) % w;
+        coord.y = (coord.y % h + h) % h;
+    }
+
+    var xi = Math.floor(coord.x); //integral part
+    var yi = Math.floor(coord.y);
+
+    if (!Number.isFinite(xi) || !Number.isFinite(yi)) return nada;
+
+    var rgba = [0, 0, 0, 0];
+    if (interpolate) {
+        var i, j;
+
+        var xf = coord.x - xi; //fractional part
+        var yf = coord.y - yi;
+
+        var pixels = readPixelsIndirection(img, xi, yi, 2, 2);
+
+        //modify pixels for boundary cases:
+        if (repeat) { //read pixels at boundary seperately
+            if (xi === w - 1 || yi === h - 1) {
+                var p10 = readPixelsIndirection(img, (xi + 1) % w, yi, 1, 1);
+                var p01 = readPixelsIndirection(img, xi, (yi + 1) % h, 1, 1);
+                var p11 = readPixelsIndirection(img, (xi + 1) % w, (yi + 1) % h, 1, 1);
+                pixels = pixels.slice(0, 4).concat(p10).concat(p01).concat(p11);
+            }
+        } else { //clamp to boundary
+            if (xi === -1 && xf >= 0.5)
+                for (i = 0; i < 4; i++)
+                    for (j = 0; j < 2; j++) pixels[8 * j + i] = pixels[8 * j + i + 4];
+            if (xi === w - 1 && xf < 0.5)
+                for (i = 0; i < 4; i++)
+                    for (j = 0; j < 2; j++) pixels[8 * j + i + 4] = pixels[8 * j + i];
+            if (yi === -1 && yf >= 0.5)
+                for (i = 0; i < 8; i++) pixels[i] = pixels[i + 8];
+            if (yi === h - 1 && yf < 0.5)
+                for (i = 0; i < 8; i++) pixels[i + 8] = pixels[i];
```

Comment:
The intention of not mixing the pixels with [0,0,0,0] is that I tried to model the way as a image is drawn by context.drawImage: No interpolation is done at the outer half pixels at the boundary. This can be verified in `examples/148_ImageRGBA4.html`.
I'll replace the code by yours. Thanks :-)